### PR TITLE
Add option to disable using gpu and change to overlap calcs

### PR DESCRIPTION
**Pull request by GrouchyPenguin**
If compiled with GPU, this binary will use a GPU if it finds one.  This allows us to disable using the GPU if desired.  Also changes overlap to a fixed pixel value.

### CODE REVIEW

File: tensorflow/cc/inference.cpp [hunk 1]
```diff
@@ -259,7 +260,26 @@ Status LoadGraph(const string &graph_file_name,
     return tensorflow::errors::FailedPrecondition("Failed to parse compute graph at '", graph_file_name, "'");
   }
 
+  // TODO dwh: not the following code should work but there is a bug that breaks it in tensorflow
```
Comment (bitweeder) [inline tensorflow/cc/inference.cpp]: `not`->`note`

File: tensorflow/cc/inference.cpp [hunk 2]
```diff
@@ -683,7 +710,9 @@ int main(int argc, char *argv[]) {
 
   // First we load and initialize the model.
   std::unique_ptr<tensorflow::Session> session;
-  // TODO dwh: use gpu for session if available
+  if (!use_gpu) {
```
Comment (bitweeder) [inline tensorflow/cc/inference.cpp]: Usage clarification: is this supposed to be `(!use_gpu)` or `(use_gpu)`?

File: tensorflow/cc/inference.cpp [hunk 3]
```diff
@@ -768,7 +797,8 @@ int main(int argc, char *argv[]) {
       rightImage = orig_image_mat(rightROI);
 
       if (do_quads) {
-        sub_image_height = std::max(static_cast<int>(static_cast<float>(image_height) * overlap_fraction / 2.f), input_height);
+        sub_image_height = std::max(overlap_pixels + static_cast<int>(static_cast<float>(image_height) / 2.f), input_height);
```
Comment (bitweeder) [inline tensorflow/cc/inference.cpp]: Clarification: is `overlap_pixels` supposed to be added in before or after the division by 2?

File: tensorflow/cc/inference.cpp (hunk 2 above)
Comment (GrouchyPenguin) [reply tensorflow/cc/inference.cpp]: So if not-use_gpu, we set an environment that lists no available devices.  Conversely, if use_gpu, then we don't modify the environment and whatever devices are there are available to tensorflow

File: tensorflow/cc/inference.cpp (hunk 3 above)
Comment (GrouchyPenguin) [reply tensorflow/cc/inference.cpp]: After division

File: tensorflow/cc/inference.cpp (hunk 1 above)
Comment (GrouchyPenguin) [reply tensorflow/cc/inference.cpp]: Fixed

File: tensorflow/cc/inference.cpp (hunk 2 above)
Comment (bitweeder) [reply tensorflow/cc/inference.cpp]: Got it; I wasn’t clear what we were setting here, exactly.

**Pull request merged**